src/ray/gcs/redis_gcs_client.h

```diff
@@ -16,38 +17,31 @@ namespace gcs {
 
 class RedisContext;
 
-class RAY_EXPORT AsyncGcsClient {
+class RAY_EXPORT RedisGcsClient : public GcsClientInterface {
+  friend class ActorStateAccessor;
+
  public:
-  /// Start a GCS client with the given client ID and command type (regular or
-  /// chain-replicated). To read from the GCS tables, Connect() and then
-  /// Attach() must be called. To read and write from the GCS tables requires a
-  /// further call to Connect() to the client table.
+  /// Constructor of RedisGcsClient.
+  /// Connect() must be called(and return ok) before you call any other methods.
+  /// To read and write from the GCS tables requires a further call to Connect()
+  /// to the client table. Will fix this in next pr.
   ///
-  /// \param address The GCS IP address.
-  /// \param port The GCS port.
-  /// \param sharding If true, use sharded redis for the GCS.
-  /// \param client_id The ID to assign to the client.
-  /// \param command_type GCS command type.  If CommandType::kChain, chain-replicated
-  /// versions of the tables might be used, if available.
-  AsyncGcsClient(const std::string &address, int port, const ClientID &client_id,
-                 CommandType command_type, bool is_test_client,
-                 const std::string &redis_password);
-  AsyncGcsClient(const std::string &address, int port, const ClientID &client_id,
-                 bool is_test_client, const std::string &password);
-  AsyncGcsClient(const std::string &address, int port, CommandType command_type);
-  AsyncGcsClient(const std::string &address, int port, CommandType command_type,
-                 bool is_test_client);
-  AsyncGcsClient(const std::string &address, int port, const std::string &password);
-  AsyncGcsClient(const std::string &address, int port, bool is_test_client);
+  /// \param ClientOption Options of client, e.g. server address, is test client ...
+  RedisGcsClient(const ClientOption &option);
 
-  /// Attach this client to an asio event loop. Note that only
-  /// one event loop should be attached at a time.
-  Status Attach(boost::asio::io_service &io_service);
+  /// Connect to GCS Service. Non-thread safe.
+  /// Call this function before calling other functions.
+  ///
+  /// \return Status
+  Status Connect(boost::asio::io_service &io_service);
+
+  /// Disconnect with GCS Service. Non-thread safe.
+  /// TODO(micafan) rename if nothing related to Disconnect
```

Comment:
I don't understand what you mean by "rename if nothing related to Disconnect".
What is this method supposed to do?

Reply:
What i mean is we can use `Shutdown` instead of `Disconnect` if we don't do any disconnect action in GcsClient. Those disconnect action like "node/raylet disconnected with gcs" might wrapped in NodeStateAccessor instead of GcsClient.

Reply:
Can you refine this comment and make the meaning clearer? thanks.

Reply:
I will use Shutdown instead of Disconnect.